Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -44,8 +44,8 @@ Common::Symbol* PPCSymbolDB::AddFunction(const Core::CPUThreadGuard& guard, u32
   if (!PPCAnalyst::AnalyzeFunction(guard, start_addr, symbol))
     return nullptr;
 
-  m_functions[start_addr] = std::move(symbol);
-  Common::Symbol* ptr = &m_functions[start_addr];
+  const auto insert = m_functions.emplace(start_addr, std::move(symbol));
```

Comment:
Can this fail?  Worth using `try_emplace`?

Reply:
If it fails, it's because there's already an element in the map with the same key (an existence check is done at the beginning of the function that makes this not possible.

Reply:
Ah, I completely glossed over the check at the start.  Sorry!

Reply:
It's all good! no need to apologize